test/integration-test/TestIIRDeserializer.cpp

```diff
@@ -0,0 +1,607 @@
+  // Register the call and set it as a replacement for the next vertical region
+  target->getMetaData().addStencilCallStmt(stencilCallDeclStmt, stencilID);
+
+  auto stencilCallStatement = std::make_shared<Statement>(stencilCallDeclStmt, nullptr);
+  target->getIIR()->getControlFlowDescriptor().insertStmt(stencilCallStatement);
+
+  ///////////////// Generation of the Metadata
+
+  target->getMetaData().insertExprToAccessID(lhsTmp, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT1, inFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT2, inFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT3, inFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT4, inFieldID);
+
+  target->getMetaData().insertExprToAccessID(lhsOut, outFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT1, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT2, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT3, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT4, tmpFieldID);
+
+  target->getMetaData().setStencilname("generated");
+
+  for(const auto& MS : iterateIIROver<iir::MultiStage>(*(target->getIIR()))) {
+    MS->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+  // Iterate all statements (top -> bottom)
+  for(const auto& stagePtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    iir::Stage& stage = *stagePtr;
+    for(const auto& doMethod : stage.getChildren()) {
+      doMethod->update(iir::NodeUpdateType::level);
+    }
+    stage.update(iir::NodeUpdateType::level);
+  }
+  for(const auto& MSPtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    MSPtr->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+}
+
+static void readIIRFromFile(OptimizerContext &optimizer, std::shared_ptr<iir::StencilInstantiation>& target, std::string fname) {
+  target =
+      IIRSerializer::deserialize(fname, &optimizer, IIRSerializer::SK_Json);
+
+  //this is whats actually to be tested.
+  optimizer.restoreIIR("<restored>", target);
+}
+
+static bool compareIIRs(std::shared_ptr<iir::StencilInstantiation> lhs, std::shared_ptr<iir::StencilInstantiation> rhs) {
+  //first compare the (structure of the) iirs, this is a precondition before we can actually check the metadata / derived info
+  IIR_EARLY_EXIT((compareIIRstructures(lhs->getIIR().get(), rhs->getIIR().get())));
+
+  //then we compare the meta data
+  IIR_EARLY_EXIT((compareMetaData(lhs->getMetaData(), rhs->getMetaData())));
+
+  //and finally the derived info 
+  IIR_EARLY_EXIT((compareMetaData(lhs->getMetaData(), rhs->getMetaData())));
+  ((compareDerivedInformation(lhs->getIIR().get(), rhs->getIIR().get())));
+
+  return true;
+}
+
+TEST(IIRDeserializerTest, CopyStencil) {
+  char cCurrentPath[FILENAME_MAX];
+  getcwd(cCurrentPath, sizeof(cCurrentPath));
+  cCurrentPath[sizeof(cCurrentPath) - 1] = '\0'; /* not really required */
+  printf ("The current working directory is %s", cCurrentPath);
```

Comment:
pritnf is a bit too much c-like ;-) Anyway I'm wondering why we should print this?